### PR TITLE
INFRA-845: Added buzzfeed-y login templates.

### PR DESCRIPTION
**Pull request by yellottyellott**
Replaces the default sign in template with a more BuzzFeed-y one. This adds the assets at `static/auth_proxy` to the binary using [go-bindata](https://github.com/jteeuwen/go-bindata).

screen shot 2016-02-16 at 7 01 24 pm

@buzzfeed/platform-infra

### CODE REVIEW

File: static/auth_proxy/error.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (yellottyellott) [inline static/auth_proxy/error.html]: This is the default error template `oauth2_proxy` shipped with.
I think this is more for 500s.

**Comment by mccutchen** [comment]
@yellottyellott I have a crazy idea:  what if we a) inline the font files into the CSS and b) inline the CSS into the template and c) stop needing any special serving of static assets???

**Comment by yellottyellott** [comment]
Can you base64 inline a font into a css file? I think they're easier to work with as separate files, especially if we ever need to update `solid.css`. But we can try it out

**Comment by yellottyellott** [comment]
I had to replace all of the backticks in the css.
Ran `$ base64` on all 4 fonts and stuffed them into the css as data uris.
Your text editor might scream at you, but it works.

image 2016-02-18 at 12 24 54 pm

**Comment by mccutchen** [comment]
Hahahahahahah :fire: :fire: :fire: :fire: :fire: :fire: :fire:

**Comment by mccutchen** [comment]
@yellottyellott I guess the real question is whether this is a good idea.  I expect it'll be unpleasant if/when we have to update something in `solid.css`, but I do like that it's all one pile of self-contained gibberish.

An even better option might be to find out whether there's some centralized place we can reference solid.css and the web fonts (a CDN?  static.buzzfeed.com or something?) to avoid the insanity of embedding it all into one file _and_ avoid cluttering the auth proxy binary with extra static routes and static assets.

What do you think?  If we can't load the assets from somewhere else, which of these two solutions do you prefer @yellottyellott?

**Comment by yellottyellott** [comment]
Messing with big gibberish files isn't great.
Adding a bunch of stuff to serve static assets isn't great.
solid.buzzfeed.com doesn't allow CORS requests.
The css is broken up on buzzfeed.com, so I'm not sure if there's a simple `solid.css` I can just link in.

Let's just go with the 1 big gibberish file since it's the dead simplest. 
I'll check in the assets so we can at least recreate the gibberish if we need to.

**Comment by mccutchen** [comment]
@yellottyellott we don't need CORS requests to load CSS from another domain, do we?

**Comment by yellottyellott** [comment]
The css will load, but the fonts referenced in the css won't.

```
Font from origin 'http://solid.buzzfeed.com' has been blocked from loading by Cross-Origin Resource Sharing policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'null' is therefore not allowed access.
```

**Comment by mccutchen** [comment]
asldfhasdklfhas;fdhas;fdhas;dflhasldfhasdlhf

**Comment by mccutchen** [comment]
> Let's just go with the 1 big gibberish file since it's the dead simplest. 
> I'll check in the assets so we can at least recreate the gibberish if we need to.

:+1:

**Comment by rolyatmax** [comment]
I don't expect big `solid` updates that would end up affecting the styles for this login page. I'm fine with gibberish.

**Comment by mccutchen** [comment]
@rolyatmax FWIW, "Log in to your BuzzFeed Dashboard" is way too specific a title to embed into this binary, given that we will likely use it to protect access to services that aren't dashboards.  We can roll with it for now, but will probably need to address this in the future.

**Comment by rolyatmax** [comment]
> FWIW, "Log in to your BuzzFeed Dashboard" is way too specific a title to embed into this binary, given that we will likely use it to protect access to services that aren't dashboards.

@mccutchen Yeah - that's a good point. @jacquelineyue Is there a more general phrase we could use here?

**Comment by jacquelineyue** [comment]
How about just `Log in to BuzzFeed`? General enough? 

About Solid.css: 
@rolyatmax is right about this: 
> I don't expect big solid updates that would end up affecting the styles for this login page. 

I expect VERY little change in this login page down the road, so gibberish solution is fine by me. 

Right now there is not a single source to grab the fonts, and every project has to include a copy on its own. Inclining the font data urls seems like the only way to go.

**Comment by mccutchen** [comment]
> How about just `Log in to BuzzFeed`? General enough?

:+1: :100: :fire:

**Comment by yellottyellott** [comment]
Made the h2 more generic. 
Rebased into two commits. 
I think this is ready!

**Comment by mccutchen** [comment]
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :rocket:  :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: 
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:
:art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art: :art:

**Comment by mccutchen** [comment]
@yellottyellott oh yeah, bump version?

**Comment by yellottyellott** [comment]
bump'd!

**Comment by mccutchen** [comment]
butan'd

**Comment by rolyatmax** [comment]
Do we need to produce a new build for this? Or is it enough for me to just run chef on the dashbird boxes?

**Comment by mccutchen** [comment]
@rolyatmax I think we just need to bump the version in the `auth_proxy` chef cookbook

**Comment by rolyatmax** [comment]
https://github.com/buzzfeed/chef/pull/1855

**Pull request merged**